internal/actor/donor_provided_test.go

```diff
@@ -20,19 +21,117 @@ var address = place.Address{
 }
 
 func TestGenerateHash(t *testing.T) {
-	donor := &DonorProvidedDetails{Attorneys: Attorneys{
-		Attorneys: []Attorney{
-			{DateOfBirth: date.New("2000", "1", "2")},
+	makeDonor := func(version uint8, hash uint64) *DonorProvidedDetails {
+		return &DonorProvidedDetails{
+			HashVersion: version,
+			Hash:        hash,
+			Attorneys: Attorneys{
+				Attorneys: []Attorney{
+					{DateOfBirth: date.New("2000", "1", "2")},
+				},
+			},
+		}
+	}
+
+	// DO change this value to match the updates
+	const modified uint64 = 0x6cc76d93a3e8e7d8
+
+	// DO NOT change these initial hash values. If a field has been added/removed
+	// you will need to handle the version gracefully by modifying HashInclude and
+	// adding another testcase for the new version.
+	testcases := map[uint8]uint64{
+		0: 0x228526f2932e1744,
+		1: 0xb4184bcfa870e40a,
+	}
+
+	for version, initial := range testcases {
+		t.Run(fmt.Sprintf("Version%d", version), func(t *testing.T) {
+			donor := makeDonor(version, initial)
+			hash, _ := donor.generateHash()
+
+			assert.Equal(t, donor.Hash, hash)
+			assert.False(t, donor.HashChanged())
+
+			donor.Attorneys.Attorneys[0].DateOfBirth = date.New("2001", "1", "2")
+			assert.True(t, donor.HashChanged())
+
+			err := donor.UpdateHash()
+			assert.Nil(t, err)
+			assert.Equal(t, modified, donor.Hash)
+			assert.Equal(t, uint8(1), donor.HashVersion)
+		})
+	}
+}
+
+func TestGenerateHashVersionTooHigh(t *testing.T) {
+	donor := &DonorProvidedDetails{
+		HashVersion: 2,
+		Attorneys: Attorneys{
+			Attorneys: []Attorney{
+				{DateOfBirth: date.New("2000", "1", "2")},
+			},
 		},
-	}}
-	hash, err := donor.GenerateHash()
-	assert.Nil(t, err)
-	assert.Equal(t, uint64(0x228526f2932e1744), hash)
-
-	donor.Attorneys.Attorneys[0].DateOfBirth = date.New("2001", "1", "2")
-	hash, err = donor.GenerateHash()
-	assert.Nil(t, err)
-	assert.Equal(t, uint64(0xf07785fd2f4548aa), hash)
+	}
+
+	_, err := donor.generateHash()
+	assert.Error(t, err)
+}
+
+func TestGenerateCheckedHash(t *testing.T) {
+	makeDonor := func(version uint8, hash uint64) *DonorProvidedDetails {
+		return &DonorProvidedDetails{
+			CheckedHashVersion: version,
+			CheckedHash:        hash,
+			Attorneys: Attorneys{
+				Attorneys: []Attorney{
+					{DateOfBirth: date.New("2000", "1", "2")},
+				},
+			},
+		}
+	}
+
+	// DO change this value to match the updates
+	const modified uint64 = 0x91100db15d995176
+
+	// DO NOT change these initial hash values. If a field has been added/removed
+	// you will need to handle the version gracefully by modifying
+	// CheckedHashInclude and adding another testcase for the new version.
```

Comment:
Should this be `CheckedHash`? Or is it referring to `toCheck.HashInclude()`?

Reply:
yep, the latter. I'll update